### PR TITLE
bpo-34187: _WindowsConsoleIO rises correct exception

### PR DESCRIPTION
**Pull request by augustogoulart**
`_WindowsConsoleIO` implementation of fileno should raise a `ValueError`
instead of `io.UnsupportedOperation` if the internal handle value is
INVALID_HANDLE_VALUE.

Previous issue that originated issue34187:
https://bugs.python.org/issue34115

https://bugs.python.org/issue34187

### CODE REVIEW

**Comment by augustogoulart** [comment]
@taleinat

File: Modules/_io/winconsoleio.c [hunk 1]
```diff
@@ -474,16 +475,24 @@ static PyObject *
 _io__WindowsConsoleIO_fileno_impl(winconsoleio *self)
 /*[clinic end generated code: output=006fa74ce3b5cfbf input=079adc330ddaabe6]*/
 {
-    if (self->fd < 0 && self->handle != INVALID_HANDLE_VALUE) {
+    if (self->handle == INVALID_HANDLE_VALUE)
```
Comment (taleinat) [inline Modules/_io/winconsoleio.c]: This `if` needs curly braces.

**Review by taleinat — CHANGES_REQUESTED** [review]
Thanks for making this PR!

ISTM that the potential issue in `internal_close` is separate from the main issue regarding `fileno`. I recommend removing the `internal_close` change from this PR, and possibly creating a separate PR for it.

Regarding the `fileno` issue, please add a test that reproduces the relevant bug.

**Comment by augustogoulart** [comment]
I have made the requested changes; please review again.

File: Lib/test/test_winconsoleio.py [hunk 2]
```diff
@@ -125,6 +125,14 @@ def test_write_empty_data(self):
         with ConIO('CONOUT$', 'w') as f:
             self.assertEqual(f.write(b''), 0)
 
+    def test_fileno_raises_correct_exception(self):
```
Comment (taleinat) [inline Lib/test/test_winconsoleio.py:128]: I suggest replacing "correct_exception" with something about the file being closed.

**Review by taleinat — CHANGES_REQUESTED** [review]
ISTM this change does more than just raise a different exception, since it also makes sure to close the file handle in a certain case. I'd like to see a test for that too.

This will need a NEWS entry; install [blurb](https://pypi.org/project/blurb/) and run `blurb add` to create one.

Also a minor inline comment.

**Pull request closed without merging**